### PR TITLE
Feat/search load

### PR DESCRIPTION
**Pull request by carpawell**

### CODE REVIEW

File: internal/native/client.go [hunk 1]
```diff
@@ -323,6 +323,58 @@ func (c *Client) PutContainer(params map[string]string) PutContainerResponse {
...
+
+		filters.AddFilter(flt.Key, flt.Value, op)
+	}
+
+	var prm client.PrmObjectSearch
+	prm.SetFilters(filters)
+
+	start := time.Now()
+
+	r, err := c.cli.ObjectSearchInit(c.vu.Context(), cID, c.signer, prm)
```
Comment (smallhive) [inline internal/native/client.go:354]: I would suggest adding _defer_ to close _search_, after `err` checking
```golang
	defer func() {
		_ = r.Close()
	}()
```

File: internal/native/client.go [hunk 2]
```diff
@@ -323,6 +323,58 @@ func (c *Client) PutContainer(params map[string]string) PutContainerResponse {
...
+	err = r.Iterate(func(_ oid.ID) bool {
+		objsNum++
+		return false
+	})
+	if err != nil {
+		return 0, fmt.Errorf("reading search results: %w", err)
+	}
+
+	var relativeTime time.Duration
+	if objsNum > 0 {
```
Comment (smallhive) [inline internal/native/client.go:369]: We spent some time even though we had zero objects. Don't we report this time to metrics?

**Comment by EESergey** [comment]
The search time should be calculated as the total search time in the container divided by the number of objects; it should not depend on the number of objects found.

**Comment by EESergey** [comment]
@cthulhu-rider Are you interested in the scenario when you specify a json file instead of a cid?

**Review by smallhive — APPROVED** [review]

File: internal/native/client.go [hunk 3]
```diff
@@ -323,6 +323,61 @@ func (c *Client) PutContainer(params map[string]string) PutContainerResponse {
 	return PutContainerResponse{Success: true, ContainerID: contID.EncodeToString()}
 }
 
+type filter struct {
+	Key       string
+	Operation string
+	Value     string
+}
+
+func (c *Client) Search(cnrString string, filtersJS []filter) (int, error) {
```
Comment (cthulhu-rider) [inline internal/native/client.go:332]: lets add docs at least to know what does `int` return means

Comment (cthulhu-rider) [inline internal/native/client.go:332]: why `filter` isnt exported?

File: scenarios/grpc_search.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+const container = __ENV.cid
+
+export const options = {
+    scenarios: {
+        system_write: {
+            executor: 'shared-iterations',
+            vus: __ENV.vu,
+            iterations: __ENV.i,
+            exec: 'search',
+            maxDuration: (24*365*100).toString()+"h", // default is 10m and this load is designed to be controlled by iterations only
```
Comment (cthulhu-rider) [inline scenarios/grpc_search.js:14]: doesnt zero mean inf?

File: scenarios/grpc_search.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { check } from 'k6';
+import native from 'k6/x/neofs/native';
+
+const grpc_client = native.connect(__ENV.GRPC_ENDPOINT, '', 5, 15);
```
Comment (cthulhu-rider) [inline scenarios/grpc_search.js:4]: id make consts for non-obvious literals

**Review by EESergey — DISMISSED** [review]

File: internal/native/client.go (hunk 3 above)
Comment (carpawell) [reply internal/native/client.go:332]: this doc is not expected to be read by anyone. this code is also impossible to use by a human, only as an extension inside a virtual machine

there are some well-known extensions that are "documented" via examples: https://github.com/grafana/xk6-loki/tree/5248b47b4efdd234403ba13e6f46dc173f9a142e/examples or via a readmi file (https://github.com/grafana/xk6-loki/blob/5248b47b4efdd234403ba13e6f46dc173f9a142e/README.md)

not sure if it applies to us (#85)

Comment (carpawell) [reply internal/native/client.go:332]: why it should be? it is only a container that will be used when code is translated from JS to go

File: scenarios/grpc_search.js (hunk 4 above)
Comment (carpawell) [reply scenarios/grpc_search.js:14]: idk, and it is not clear from docs: https://k6.io/docs/using-k6/k6-options/reference/#iterations
we can try what it means in practice. not sure if we can rely on it

File: scenarios/grpc_search.js (hunk 5 above)
Comment (carpawell) [reply scenarios/grpc_search.js]: dont mind

File: internal/native/client.go (hunk 3 above)
Comment (cthulhu-rider) [reply internal/native/client.go:332]: why method is exported then? im lookin at this code as a regular Go one

File: scenarios/grpc_search.js (hunk 4 above)
Comment (cthulhu-rider) [reply scenarios/grpc_search.js:16]: docs dont state this, so we cannot

File: scenarios/grpc_search.js (hunk 5 above)
Comment (cthulhu-rider) [reply scenarios/grpc_search.js]: and empty string pls, i can only imagine what does it mean

File: internal/native/client.go (hunk 3 above)
Comment (cthulhu-rider) [reply internal/native/client.go:332]: is this an autogen code? If not, any dev can read and support it. Small tip would be appreciated

**Review by smallhive — DISMISSED** [review]

File: internal/native/client.go (hunk 3 above)
Comment (carpawell) [reply internal/native/client.go:332]: > why method is exported then?

to make it possible to be called in k6 code

> im lookin at this code as a regular Go one

idk, i think this code should be considered as a private package code, it is called by an external k6 code that is never human (and that is never written based on our neofs code) and that just looks for exported fields with appropriate names. can be exported, OK, but changes nothing

Comment (carpawell) [reply internal/native/client.go]: it is not a problem to me, added a comment. now we have one documented method out of 7

**Pull request merged**